multiversx_sdk_cli/delegation/staking_provider.py

```diff
@@ -1,169 +1,347 @@
+            tx = self._factory.create_transaction_for_setting_automatic_activation(
+                sender=owner.address,
+                delegation_contract=delegation_contract
+            )
+        elif args.unset:
+            tx = self._factory.create_transaction_for_unsetting_automatic_activation(
+                sender=owner.address,
+                delegation_contract=delegation_contract
+            )
+
+        tx.nonce = int(args.nonce)
+        tx.version = int(args.version)
+        tx.options = int(args.options)
+        tx.guardian = args.guardian
+        tx.signature = bytes.fromhex(owner.sign_transaction(tx))
+
+        if args.gas_limit:
+            tx.gas_limit = int(args.gas_limit)
+
+        return tx
+
+    def get_transaction_for_redelegate_cap(self, owner: IAccount, args: Any) -> ITransaction:
+        delegation_contract = Address.new_from_bech32(args.delegation_contract)
+
+        tx = self.__get_empty_transaction()
+        if args.set:
+            tx = self._factory.create_transaction_for_setting_cap_check_on_redelegate_rewards(
+                sender=owner.address,
+                delegation_contract=delegation_contract
+            )
+        elif args.unset:
+            tx = self._factory.create_transaction_for_unsetting_cap_check_on_redelegate_rewards(
+                sender=owner.address,
+                delegation_contract=delegation_contract
+            )
+
+        tx.nonce = int(args.nonce)
+        tx.version = int(args.version)
+        tx.options = int(args.options)
+        tx.guardian = args.guardian
+        tx.signature = bytes.fromhex(owner.sign_transaction(tx))
+
+        if args.gas_limit:
+            tx.gas_limit = int(args.gas_limit)
+
+        return tx
+
+    def get_transaction_for_setting_metadata(self, owner: IAccount, args: Any) -> ITransaction:
+        delegation_contract = Address.new_from_bech32(args.delegation_contract)
+
+        tx = self._factory.create_transaction_for_setting_metadata(
+            sender=owner.address,
+            delegation_contract=delegation_contract,
+            name=args.name,
+            website=args.website,
+            identifier=args.identifier
+        )
+        tx.nonce = int(args.nonce)
+        tx.version = int(args.version)
+        tx.options = int(args.options)
+        tx.guardian = args.guardian
+        tx.signature = bytes.fromhex(owner.sign_transaction(tx))
+
+        if args.gas_limit:
+            tx.gas_limit = int(args.gas_limit)
+
+        return tx
+
+    def _parse_public_bls_keys(self, public_bls_keys: str) -> List[ValidatorPublicKey]:
+        keys = public_bls_keys.split(",")
+        validator_public_keys: List[ValidatorPublicKey] = []
+
+        for key in keys:
+            validator_public_keys.append(ValidatorPublicKey(bytes.fromhex(key)))
+
+        return validator_public_keys
+
+    def _get_public_keys_and_signed_messages(self, args: Any) -> Tuple[List[ValidatorPublicKey], List[bytes]]:
```

Comment:
Maybe some of the logic can be moved to `ValidatorsFile`? E.g. `load_secret_keys()`? Or even `load_signers()` - and the returned object (signers) allows us to both sign a message, and get the underlying public keys.

Reply:
Made a function `load_signers()` inside the `ValidatorsFile` class that returns a list of `ValidatorSigners`.